src/daemon.py

```diff
@@ -108,7 +108,13 @@ def __search_task(self, job_id: JobId) -> None:
         iterator = result["iterator"]
         logging.info(f"Iterator {iterator} contains {file_count} files")
 
-        self.db.update_job_files(job_id, file_count)
+        total_files = self.db.update_job_files(job_id, file_count)
+        if job.files_limit and total_files > job.files_limit:
+            raise RuntimeError(
+                "Exceeded the limit of files to be processed (%d). Try more specific query.",
+                job.files_limit,
```

Comment:
```suggestion
                "Too many candidates after prefiltering (limit: %s). Try a more specific query.",
                job.files_limit,
```

Reply:
Applied suggestion with small improvement (f-string instead of tuple)